src/pobox.erl

```diff
@@ -198,21 +310,38 @@ notify(cast, _Msg, _State) ->
     keep_state_and_data;
 notify({call, From}, Msg, Data) ->    
     handle_call(From, Msg, notify, Data);
-notify(info, Msg, Data) ->    
+notify(info, Msg, Data) ->
     handle_info(Msg, notify, Data).
         
 
 %% @private
-handle_call(From, {resize, NewSize}, _StateName, S=#state{buf=Buf}) ->
-    {keep_state, S#state{buf=resize_buf(NewSize,Buf)}, 
-        [{reply, From, ok}]};
 handle_call(From, {post, Msg}, StateName, S=#state{buf=#buf{max=Size, size=Size}}) ->
     gen_server:reply(From, full),
     ?MODULE:StateName(cast, {post, Msg}, S);
 handle_call(From, {post, Msg}, StateName, S) ->
     gen_server:reply(From, ok),
     ?MODULE:StateName(cast, {post, Msg}, S);
-
+handle_call(From, {resize, NewSize}, _StateName, S=#state{buf=Buf}) ->
+    {keep_state, S#state{buf=resize_buf(NewSize,Buf)}, [{reply, From, ok}]};
+handle_call(From, {give_away, Dest, DestData, Origin}, _StateName, S=#state{owner=Owner, owner_pid=OwnerPid, name=Name}) ->
+    CanUsePid = case where(Dest) of
+        DestPid when is_pid(DestPid) ->
+            is_process_alive(DestPid) and (Origin =:= OwnerPid) and (DestPid =/= OwnerPid);
+        _ -> false
+    end,
+    case CanUsePid of
+        true ->
+          case send_ownership_transfer(Owner, Dest, DestData, Name) of
```

Comment:
What if we get exit from our parent while this is happening? 

Dest will receive a transfer message but we will terminate because of the exit message. To prevent this we need to change the parent here (by changing the ancestors in pdict) or switch to monitor.